### ISSUE TITLE
Not working on anything other than Firefox

### ISSUE THREAD

**Issue by Eddippla** [Code-Break0/SecureAuthSystem#3]
https://github.com/Code-Break0/SecureAuthSystem/issues/2, sorry for making this new ticket but, you could see my version of this system at: https://eddippla.me, it should show you it doesn't work, I really need this to work on other browsers as most people don't use firefox. Also, for you saying that it might be an outdated version, that's really unlikely as I set it up today, and haven't changed anything in the code. If you need me to send you any of my code I can send you it

**Comment by Eddippla** [comment]
Hey! I found the issue.
The code below was the issue.
<base href="<?php echo dirname($_SERVER['PHP_SELF']); ?>/" />
                                                                                                                           ^
                                                                                                                           this slash made it not work.

**Comment by Code-Break0** [comment]
Sorry about that I didn't mean to shut you out. I just like to keep the issues section for problems with the code, not different implementations. Since the code works in chrome for me there must be an issue with your implementation. However, I will leave this issue open until we can resolve your problem. I am reluctant to click on the link hope you can understand but if you can email me at codebreak05@gmail.com that would be great thanks

**Comment by Code-Break0** [comment]
Ah I see that is an important thing to note... I guess for my system localhost the extra trailing '/' is not an issue. In fact this is usually the case. However your system must not tolerate trailing /

**Comment by Code-Break0** [comment]
In fact on my system removing the trailing / actually breaks the code... strange. In the future I will avoid the <base> tag since it is kind of a undesired solution to fixing the link-rewriting

**Comment by Eddippla** [comment]
Actually, it might be a problem for me since I removed the .htaccess file because rewrites wouldn't work for me.
It's now completely working for me.

**Comment by Code-Break0** [comment]
Ah ok that is a very important detail thank you for mentioning that. Removing the .htaccess file has the potential to break many things. Out of curiosity how are you hosting your site? Are you using Apache on shared hosting or something else?

**Comment by Eddippla** [comment]
Oh, I run my own webserver, (apache) I have also fixed all the things that removing the .htaccess file would break, email verification, passwords resets, etc.

**Comment by Eddippla** [comment]
Also, do you know how to edit the login system to use people's real IP addresses for failed login attempts, as I'm using cloudflare, and I don't want cloudflare IP addresses to be blocked from logging in.

**Comment by Code-Break0** [comment]
Take a look at this https://stackoverflow.com/questions/3003145/how-to-get-the-client-ip-address-in-php

I wouldn't rely on the client's IP for anything I just added it as a fun little thing.

**Comment by sallyu2021** [comment]
I have read your conversations above. I have the same problem. So I deleted the "/" in the file but it does not work. All background did not show in Google and Edge. In fact, it was perfectly fine when I use XAMPP and localhost. Then I uploaded to cPanel of my web hosting company. It did work. The registration form only works in Firefox. Safari said I have inserted invalid address. My web hosting company said it was the coding problem, and they are not helping me solve the problem. Pls advise.

Thank you!
